atividade03/src/main/java/br/edu/uepb/eda/atividade03/PilhaComLista.java

```diff
@@ -0,0 +1,45 @@
+package br.edu.uepb.eda.atividade03;
+
+import br.edu.uepb.eda.atividade03.ListaEncadeada;
+import br.edu.uepb.eda.atividade03.Pilha_IF;
+
+public class PilhaComLista implements Pilha_IF {
+    public ListaEncadeada lista = new ListaEncadeada();
+    @Override
+    public void push(Integer element) throws Exception {
+        if (isFull()) {
+            throw new Exception("Pilha cheia.");
+        }
+        lista.insert(element);
+    }
+
+    @Override
+    public Integer pop() throws Exception {
+        if (isEmpty()) {
+            throw new Exception("Pilha vazia.");
+        }
+        int top = lista.valor;
+        lista.valor = lista.prox.valor;
+        lista.prox = lista.prox.prox;
+        return top;
+    }
+
+    @Override
+    public Integer top() throws Exception {
+        if (isEmpty()) {
+            throw new Exception("Pilha vazia.");
+        }
+        return lista.valor;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return lista.valor == null;
+    }
+
+    @Override
+    public boolean isFull() {
+        return lista.size()==10;
```

Comment:
Apenas um ponto de atenção, cuidado com magic number ou valores constantes soltos pelo código, o ideal é levar esse valor para uma variável que represente o contexto daquela informação, por exemplo:

primeiro declara a variável constante com um nome que descreve o valor que ela receber
```
private final int MAX_ELEMENT = 10;
```

```suggestion
		if (l.size() >= this.MAX_ELEMENT) {
```